### PR TITLE
Vignette love

### PR DESCRIPTION
**Pull request by mattansb**
#133

### CODE REVIEW

**Comment by mattansb** [comment]
@strengejacke @DominiqueMakowski 
Any idea why I'm getting this:
```
❯ checking for code/documentation mismatches ... WARNING
  Variables with usage in documentation object 'hardlyworking' but not in code:
    ‘hardlyworking’
```

as far as I can see I've done it right (according to Hadley's book...)

**Comment by mattansb** [comment]
@strengejacke Help on this annoying warning? 🤪

File: NEWS.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 ## Bug fixes
 
+- Setting `two_sd = TRUE` in `standardize()` and `standardize_parameters()` (correctly) on uses 2-SDs of the predictors (and not the response). 
 - `standardize_info()` / `standardize_parameters(method = "posthoc")` work for zero-inflated models ( #135 )
```
Comment (mattansb) [inline NEWS.md:18]: @DominiqueMakowski FYI This wasn't working 100% correctly (especially with `method = "refit"`), probably for a long time - all fixed now. Just thought you'd like to know (:

**Comment by mattansb** [comment]
image
 
@strengejacke Any idea what this is? Locally this vignette runs just fine...

```
--- re-building ‘standardize_parameters.Rmd’ using rmarkdown
Quitting from lines 115-118 (standardize_parameters.Rmd) 
Error: Error: processing vignette 'standardize_parameters.Rmd' failed with diagnostics:
missing value where TRUE/FALSE needed
--- failed re-building ‘standardize_parameters.Rmd’
```

**Comment by mattansb** [comment]
Error starts after this:

https://github.com/easystats/effectsize/pull/137/commits/08cf48ff178166ffeba82f5f3d1dae0f21bcdb92

But why??

**Comment by DominiqueMakowski** [comment]
let me try running the checks locally

**Comment by DominiqueMakowski** [comment]
```
--- re-building 'standardize_parameters.Rmd' using rmarkdown
   Quitting from lines 115-120 (standardize_parameters.Rmd) 
   Error: processing vignette 'standardize_parameters.Rmd' failed with diagnostics:
   missing value where TRUE/FALSE needed
```

I confirm it's a legit thing 😅

**Comment by DominiqueMakowski** [comment]
but indeed the chunk works when run directly 🤔

**Comment by strengejacke** [comment]
I have to install the dev-branch, right?

File: R/eta_squared_posterior.R [hunk 2]
```diff
@@ -152,7 +152,7 @@ eta_squared_posterior.brmsfit <- eta_squared_posterior.stanreg
     # if a contrast has negative and positive values, it is assumed to be one of:
     # "contr.sum", "contr.helmert", "contr.poly", "contr.bayes"
     factors_centered <- sapply(X[, factors, drop = FALSE],
-                               function (xi) any(contrasts(xi) < 0) & any(contrasts(xi) > 0))
+                               function(xi) any(contrasts(xi) < 0) & any(contrasts(xi) > 0))
```
Comment (mattansb) [inline R/eta_squared_posterior.R:155]: Yes, very important! 🤣

**Comment by strengejacke** [comment]
The function calls are extremely nested, so I can't figure out the source of this failing check / error in the vignette. I suppose one of the if-statements somewhere down in `standardize_info()` might fail, because the elements from `parameters_type()` may contain missing values...

**Comment by strengejacke** [comment]
Damn, I'm so good... 🚀

**Comment by mattansb** [comment]
Genius!

---

Adding the Bayes vignette still breaks the standardize_parameters vignette that calls `standardize_info()` (how? why? makes no sense...) - will investigate this again tomorrow, on a separate PR

**Comment by strengejacke** [comment]
I'm again so damn genius! :-D

``` r
library(effectsize)
options(contrasts = c('contr.sum', 'contr.poly'))
mtcars <- datasets::mtcars
mtcars$am <- factor(mtcars$am, labels = c("Manual", "Automatic"))
m <- lm(mpg ~ am, data = mtcars)
standardize_parameters(m, method = "smart")
#> Error in if (types$Link[types$Parameter == var] == "Difference") {: missing value where TRUE/FALSE needed
```

<sup>Created on 2020-09-28 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by strengejacke** [comment]
It's the contrast-setting in the Bayes vignette that breaks the code. But I did not dig deeper into it to know why...

**Comment by strengejacke** [comment]
Since it passes all checks w/o Bayes vignette, it _must_ have to do with some "settings". I thought that there were some relics in the dataset manipulation, but then I stumbled over the contrast-code and just tried.

_Looks_ like this might be the reasons.

**Comment by mattansb** [comment]
Bloody amazing!

Will investigate further tomorrow and see why that happens - thanks!

--
Mattan S. Ben-Shachar, PhD student
Department of Psychology & Zlotowski Center for Neuroscience
Ben-Gurion University of the Negev
The Developmental ERP Lab

**Pull request merged**